src/main/java/io/airlift/slice/Slice.java

```diff
@@ -888,7 +888,7 @@ public Slice slice(int index, int length)
 
     public int indexOfByte(int b)
     {
-        b &= 0xFF;
+        checkArgument((b >> Byte.SIZE) == 0, "byte value out of range");
```

Comment:
if the advantage of this compare to `0 <= b && b < 256`?

Reply:
we should provide an overload that takes `byte`

Reply:
There’s no branching, but it’s hard to say if it’s actually faster. This is how it’s done in Guava.

Reply:
Good idea, added an overload